apache/couchdb-fauxton
# [partitioned dbs] Updates to index editor and index clone modal

## Overview

Seventh (and last 😀) in a series of PRs that will be submitted in support of the new user-defined partitioned databases feature (apache/couchdb#1605).

Summary:

 - Add 'Partitioned' checkbox to Index Editor
  - Checkbox is read-only for existing ddocs and is updated to reflect the selected ddoc
  - Checkbox is editable when choosing to create a new ddoc
  - CUSTOM reducer option is not shown when editing/creating partitioned views
 - Add 'Partitioned' checkbox to Clone Index modal

## Testing recommendations

On a partitioned database: 
- Click on 'New view'
   - Select 'New Document', name the new ddoc, and make sure 'Partitioned' is checked
   - Verify the 'CUSTOM' reducer is not listed
   - Click the button to create the view
   - Verify the new view is partitioned
- Use the menu to clone the view you just created
  - Select 'New Document', name the new ddoc and view, and uncheck 'Partitioned'
   - Click the button to clone the view
   - Verify the new view is global
   - Verify the original view still exists
- Click on 'New view'
   - Select 'New Document', name the new ddoc, and uncheck 'Partitioned' 
   - Verify the 'CUSTOM' reducer is listed
   - Click the button to create the view
   - Verify the new view is global

## Related Pull Requests

apache/couchdb#1605

## Checklist
